### DISCUSSION TITLE
Plugin snmp doesn't find snmpget on SLES

### DISCUSSION THREAD

**Discussion by mdicss** [Linuxfabrik/monitoring-plugins#704] [Q&A] [answered]
Hi
I would like to use plugin snmp (compiled version ) on SLES 15.4. net-snmp is installed. We have:

`# which snmpget`
`/usr/bin/snmpget`

`# /usr/bin/snmpget -V`
`NET-SNMP version: 5.9.3`

When I call snmp, I get a message, that snmpget is not found:
`./snmp -H xx.yy.zz.uu`
 `snmpget not found`

Any idea how to fix this?

Thanks for any hints, regards
Matthias

**Comment by NavidSassan** [comment]
can you send the output of `command -v snmpget; echo $?`? this is how the plugin checks if it can find `snmpget`, see https://github.com/Linuxfabrik/monitoring-plugins/blob/61a32aabe174b83847a42b4c45d93be00a6e5fea/check-plugins/snmp/snmp#L201

**Reply by mdicss** [reply]
Looks ok?
`# command -v snmpget; echo $?`
`/usr/bin/snmpget`
`0`

**Comment by markuslf** [comment]
Seems that the user under which the plugin is running does not have access to the `snmpget` command. Try using `sudo -u icinga snmpget`.

**Reply by mdicss** [reply]
Hi Markus

Thanks a lot for your fast and good help!
I narrowed down the problem in the meantime. In CentOS which we used until now, there is a shell skript '/bin/command', which calls the shell command 'command'.

```
# cat /bin/command
#!/bin/sh
builtin command "$@"
```

This skript is missing in our sles 15.4 installation. So I copied it from CentOS to Sles and it now works.
As searching for command 'command' in the net is a bit a problem, I did not yet find, why the skript is missing in sles.
Another aspect of the problem is, that the user icinga is a 'nologin'-user. So icinga seems not to run inside (ba)sh as (ba)sh has the builtin command 'command'.

Maybe you can change the test in your skripts?

Regards
Matthias

**Comment by markuslf** [comment, marked as answer]
Great, thanks for your tests. So the solution is simple: We stop checking for the existence of `snmpget`, because when it is called and if it is missing, we later already get a meaningful error message like `OS Error "2 No such file or directory: 'snmpget'"`.
